### PR TITLE
PUE EA license mocking and refined/extended examples

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/71337

This PR extends and refines the examples to add the `origin` attribute in the responses (see TEC PR for the ticket) and mocks the license too

### CODE REVIEW

File: tests/_support/ea-server-mocker/ea-service-mocker.php [hunk 1]
```diff
@@ -41,76 +42,84 @@ class Tribe__Events__Aggregator_Mocker {
...
 	/**
 	 * Sweeps and cleans the mess we made.
 	 */
-	public static function deactivate() {
-		foreach ( self::$option_providers as $provider ) {
-			$provided = call_user_func( array( $provider, 'provides_options' ) );
-			foreach ( $provided as $option ) {
-				delete_option( $option );
+	public static function deactivate()
+	{
```
Comment (bordoni) [inline tests/_support/ea-server-mocker/ea-service-mocker.php]: Don't break line

File: tests/_support/ea-server-mocker/ea-service-mocker.php [hunk 2]
```diff
@@ -41,76 +42,84 @@ class Tribe__Events__Aggregator_Mocker {
...
+		foreach (self::$option_providers as $provider) {
+			$provided = call_user_func(array($provider, 'provides_options'));
```
Comment (bordoni) [inline tests/_support/ea-server-mocker/ea-service-mocker.php]: Spacing.

File: tests/_support/ea-server-mocker/ea-service-mocker.php [hunk 3]
```diff
@@ -41,76 +42,84 @@ class Tribe__Events__Aggregator_Mocker {
...
 	 * Sweeps and cleans the mess we made.
 	 */
-	public static function deactivate() {
-		foreach ( self::$option_providers as $provider ) {
-			$provided = call_user_func( array( $provider, 'provides_options' ) );
-			foreach ( $provided as $option ) {
-				delete_option( $option );
+	public static function deactivate()
+	{
+		foreach (self::$option_providers as $provider) {
```
Comment (bordoni) [inline tests/_support/ea-server-mocker/ea-service-mocker.php]: Spacing

File: tests/_support/ea-server-mocker/ea-service-mocker.php [hunk 4]
```diff
@@ -41,76 +42,84 @@ class Tribe__Events__Aggregator_Mocker {
...
+			foreach ($provided as $option) {
+				delete_option($option);
 			}
 		}
-		delete_option( 'ea_mocker-enable' );
+		delete_option('ea_mocker-enable');
 	}
 
-	public function autoload( $class ) {
+	public function autoload($class)
```
Comment (bordoni) [inline tests/_support/ea-server-mocker/ea-service-mocker.php]: Spacing

File: tests/_support/ea-server-mocker/ea-service-mocker.php [hunk 5]
```diff
@@ -41,76 +42,84 @@ class Tribe__Events__Aggregator_Mocker {
...
+	{
```
Comment (bordoni) [inline tests/_support/ea-server-mocker/ea-service-mocker.php]: Break line

File: tests/_support/ea-server-mocker/src/Service_Options.php [hunk 6]
```diff
@@ -257,49 +262,78 @@ public function __construct() {
...
 			'ea_mocker-origins-mock_response',
 			'ea_mocker-import-mock_response',
 			'ea_mocker-import-post_import_mock_response',
-		);
+		) );
 	}
 
-	public function fields() {
+	public function fields()
+	{
```
Comment (bordoni) [inline tests/_support/ea-server-mocker/src/Service_Options.php]: Don't break line

File: tests/_support/ea-server-mocker/src/Service_Options.php [hunk 7]
```diff
@@ -257,49 +262,78 @@ public function __construct() {
...
+	public function hook()
+	{
 		add_filter( 'ea_mocker-settings', array( $this, 'settings' ) );
 		add_action( 'ea_mocker-options_form', array( $this, 'fields' ) );
 	}
 
-	public function settings() {
-		return array(
+	public function settings( array $settings = array() )
+	{
```
Comment (bordoni) [inline tests/_support/ea-server-mocker/src/Service_Options.php]: Don't break line

File: tests/_support/ea-server-mocker/src/Service_Options.php [hunk 8]
```diff
@@ -257,49 +262,78 @@ public function __construct() {
...
 		return array(
 			'ea_mocker-origins-mock_response',
 			'ea_mocker-import-mock_response',
 			'ea_mocker-import-post_import_mock_response',
 		);
 	}
 
-	public function hook() {
+	public function hook()
+	{
```
Comment (bordoni) [inline tests/_support/ea-server-mocker/src/Service_Options.php]: Don't break line

File: tests/_support/ea-server-mocker/src/Service_Options.php [hunk 9]
```diff
@@ -257,49 +262,78 @@ public function __construct() {
...
+				) ),
+		) );
 	}
 
 	/**
 	 * Returns an array of options the class uses.
 	 */
-	public static function provides_options() {
+	public static function provides_options()
+	{
```
Comment (bordoni) [inline tests/_support/ea-server-mocker/src/Service_Options.php]: Don't break line

File: tests/_support/ea-server-mocker/src/Service_Options.php [hunk 10]
```diff
@@ -1,254 +1,259 @@
 <?php
 
 
-class Tribe__Events__Aggregator_Mocker__Service_Options implements Tribe__Events__Aggregator_Mocker__Option_Provider_Interface {
+class Tribe__Events__Aggregator_Mocker__Service_Options implements Tribe__Events__Aggregator_Mocker__Option_Provider_Interface
+{
```
Comment (bordoni) [inline tests/_support/ea-server-mocker/src/Service_Options.php]: Don't break line

File: tests/_support/ea-server-mocker/src/License.php [hunk 11]
```diff
@@ -0,0 +1,44 @@
+<?php
```
Comment (bordoni) [inline tests/_support/ea-server-mocker/src/License.php]: Spacing and Don't break line on this file.

File: tests/_support/ea-server-mocker/ea-service-mocker.php [hunk 12]
```diff
@@ -41,76 +42,84 @@ class Tribe__Events__Aggregator_Mocker {
...
+		if (strpos($class, $prefix) === 0) {
+			$class_path = str_replace(array($prefix, '__'), array('', DIRECTORY_SEPARATOR), $class);
 			/** @noinspection PhpIncludeInspection */
-			require dirname( __FILE__ ) . '/src/' . $class_path . '.php';
+			require dirname(__FILE__) . '/src/' . $class_path . '.php';
 		}
 	}
 
-	public function mock() {
+	public function mock()
```
Comment (bordoni) [inline tests/_support/ea-server-mocker/ea-service-mocker.php]: review Spacing and Breaklines on this file.

**Comment by lucatume** [comment]
I'm having a bad code styling day... should be good

**Pull request merged**